spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/RemoteTokenServices.java

```diff
@@ -113,7 +113,7 @@ public OAuth2Authentication loadAuthentication(String accessToken) throws Authen
 		}
 
 		// gh-838
-		if (!Boolean.TRUE.equals(map.get("active"))) {
+		if (!Boolean.TRUE.toString().equals(map.get("active"))) {
```

Comment:
@Clement-Leopold With this change, `"true"` would work, but `true` would not. The provider may return the `active` attribute as either a JSON boolean or string. We need to account for both types.

Can you please update and add a test as well?

Reply:
@jgrandja  I've made some updates as you requested. thanks!